aopy/analysis/base.py

```diff
@@ -2316,7 +2317,159 @@ def windowed_xval_lda_wrapper(data, labels, samplerate, lags=3, nfolds=5, regula
         return decoding_accuracy, time_axis, cm
     else:
         return decoding_accuracy, time_axis
+
+def xval_lda_resample_wrapper(data, labels, min_trial, cond_mask=None, single_decoding=True, min_unit=None, shuffle=False, nfolds=5,\
+                               replacement=False, regularization='auto', lda_model=None, return_weights=False, seed=None):
+    '''
+    Perform an n-fold cross-validation with Linear Discriminant Analysis (LDA) using a single unit or multiple units.
+    This functions extracts trials and/or units randomly based on the random seed with/without replacement.
+    The number of trials per target is controlled to be the same across targets
+    You can repeatedly perform this function by changing random seed to estimate the resampling distribution
+
+    Args:
+        data (nunit, ntr): neural data
+        labels (ntr): the labels for each trial.
+        min_trial (int): the minimum number of trials for each label to be extracted
+                                if this is 20, 20*(the number of unique labels) trials are extracted
+        cond_mask (ntr): a boolean array. This is a mask to extract trials that satisfies a certain condition
+                                (ex. movement onset is more than a certain threshold)
+        single_decoding (bool, optional): If True, LDA is performed using single unit activity separately (default is True)
+        min_unit (int, optional): the number of units used for decoding. This is used only when single_decoding is False.
+                                if min_unit is None, decoding is perfomed using all units (default is None)
+        shuffle (bool, optional): whether to shuffle labels or not (default is False)
```

Comment:
Can you rename this to 'shuffle_labels'? To be more specific that units or trials are not shuffled.